### PR TITLE
react-accordion

### PR DESCRIPTION
**Pull request by TristanChao**

### CODE REVIEW

File: react-accordion/my-app/src/Accordion.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  topics: Topic[];
+};
+export function Accordion({ topics }: Props) {
+  const [displayId, setDisplayId] = useState<number>();
+
+  const topicBoxArray: JSX.Element[] = [];
+
+  for (let i = 0; i < topics.length; i++) {
+    const header = (
+      <TopicHeader
```
Comment (RobertGardner) [inline react-accordion/my-app/src/Accordion.tsx:20]: I would find the code more logical if a Topic component contained both the header and the content.

File: react-accordion/my-app/src/TopicContent.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+type Props = {
+  isDisplayed: boolean;
+  content: string;
+};
+export function TopicContent({ isDisplayed, content }: Props) {
+  return isDisplayed ? <div>{content}</div> : undefined;
```
Comment (RobertGardner) [inline react-accordion/my-app/src/TopicContent.tsx:6]: JSX should be `null` if you don't want it to display. `undefined` works, but the spec says it should be `null`.

File: react-accordion/my-app/src/TopicHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+};
+export function TopicHeader({
+  id,
+  currentDisplayId,
+  title,
+  onHeaderClick,
+}: Props) {
+  return (
+    <div
+      onClick={() => onHeaderClick(currentDisplayId === id ? undefined : id)}
```
Comment (RobertGardner) [inline react-accordion/my-app/src/TopicHeader.tsx:15]: I think you want to process clicks even if the card is currently displayed. If you click the one that is currently displayed, it should close. You should fix this.

**Review by RobertGardner — CHANGES_REQUESTED** [review]
One minor fix needed so that it matches the mock-up.

**Comment by TristanChao** [comment]
Recording 2024-09-03 at 09 35 36

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**